packages/amplify-provider-awscloudformation/src/__tests__/system-config-manager.test.ts

```diff
@@ -35,6 +35,22 @@ describe('profile tests', () => {
     expect(getProfileCredentials_mock).toHaveBeenCalledTimes(0);
   });
 
+  it('should fail to return profiled aws credentials', async () => {
+    fs_mock.readFileSync.mockImplementationOnce(() => {
+      return '[fake]\nmalformed_key_id=fakeAccessKey\nmalformed_secret_access_key=fakeSecretKey\n';
+    }).mockImplementationOnce(() => {
```

Comment:
Why does this need to be mocked twice?

Reply:
It's being called twice by the current code so if I only mock it once the second call returns undefined on the file and it fails, I didn't want to mock it an indefinite number of times since I'd like to catch that behavior if more calls happen in the future

I can go back and figure out why it's calling twice and fix that if we want to tack that on

Reply:
Ideally we would fix the fact that the files are read twice, but if it's too big a task to fix, then I think it's fair to say it's out of scope.

Reply:
I'm inclined to leave it, I skimmed through the functions and my first impression is that while the double read is not strictly necessary for the test, I think it has use for some file configurations in practice. I think it's because one is meant to read .aws/config and the other .aws/credentials